### ISSUE TITLE
The number syntax in jq is different from JavaScript 

### ISSUE THREAD

**Issue by hexagonrecursion** [jqlang/jq#2238]
**Describe the bug**
[The manual](https://github.com/stedolan/jq/blob/80052e5275ae8c45b20411eecdd49c945a64a412/docs/content/manual/manual.yml#L521-L522) says that numbers in the jq language are written the same way as in JavaScript, but this is not the case. 

**To Reproduce**
jq fails to parse the numeric literal separator:
```
$ node -p '123_456'
123456
$ jq -n '123_456'
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
123_456   
jq: 1 compile error
```

jq fails to parse binary, octal and hexadecimal literals:
```
$ node -p '0b111'; node -p '0o111'; node -p '0x111'
7
73
273
$ jq -n '0b111'; jq -n '0o111'; jq -n '0x111'
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
0b111 
jq: 1 compile error
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
0o111 
jq: 1 compile error
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
0x111 
jq: 1 compile error
```

**Expected behavior**
The misleading documentation should be changed.

**Environment (please complete the following information):**
```
$ jq --version
jq-1.6
$ cat /etc/fedora-release 
Fedora release 32 (Thirty Two)
```

**Additional context**
https://tc39.es/ecma262/#sec-literals-numeric-literals

**Comment by Pysis868** [comment]
Also, bare hex notation (note in string literal quotes) gives `parse error: Invalid numeric literal at line #, column #`.

**Comment by pkoppstein** [comment]
Evidently "JSON" was intended, rather than "JavaScript". It was no doubt some kind of mixup/lapse/neural misfiring/... -- you choose.

**Comment by itchyny** [comment]
Since the documentation referring to JavaScript was corrected by #2645, closing the issue.
